tests/Gelf/Test/Transport/RetryTransportWrapperTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+
+namespace Gelf\Test\Transport;
+
+
+use Gelf\Message;
+use Gelf\TestCase;
+use Gelf\Transport\AbstractTransport;
+use Gelf\Transport\RetryTransportWrapper;
+use \PHPUnit_Framework_MockObject_MockObject as MockObject;
+
+class RetryTransportWrapperTest extends TestCase
+{
+    /**
+     * @var Message
+     */
+    private $message;
+
+    /**
+     * @var AbstractTransport|MockObject
+     */
+    private $transport;
+
+    /**
+     * @var RetryTransportWrapper
+     */
+    private $wrapper;
+
+    /**
+     * @var string
+     */
+    private $successValue;
+
+    public function setUp()
+    {
+        $this->message = new Message();
+        $this->transport = $this->buildTransport();
+        $this->wrapper   = new RetryTransportWrapper($this->transport);
+        $this->successValue = "HTTP/1.1 202 Accepted\r\n\r\n";
+    }
+
+    public function testSendFailOnce()
+    {
+        $expectedException = new \RuntimeException();
+
+        $this->transport->expects($this->exactly(2))
+            ->method('send')
+            ->with($this->message)
+            ->will($this->onConsecutiveCalls(
+                $this->throwException($expectedException),
+                $this->returnValue($this->successValue)
+            ));
+
+        $bytes = $this->wrapper->send($this->message);
+
+        $this->assertEquals($this->successValue, $bytes);
+    }
+
+    public function testSendFail()
+    {
+        $expectedException = new \RuntimeException();
+
+        $this->transport->expects($this->exactly(2))
+            ->method('send')
+            ->with($this->message)
+            ->will($this->onConsecutiveCalls(
+                $this->throwException($expectedException)));
+
+        $bytes = $this->wrapper->send($this->message);
+
+        $this->assertNotEquals($this->successValue, $bytes);
+    }
+
+    public function testSendSuccess()
+    {
+        $this->transport->expects($this->once())
+            ->method('send')
+            ->with($this->message)
+            ->will($this->returnValue($this->successValue));
+
+        $bytes = $this->wrapper->send($this->message);
+
+        $this->assertEquals($this->successValue, $bytes);
+    }
+
+    /**
+     * @return MockObject|AbstractTransport
+     */
+    private function buildTransport()
+    {
+        return $this->getMockForAbstractClass("\\Gelf\\Transport\\AbstractTransport");
```

Comment:
Can you use the class keyword instead:

```suggestion
        return $this->getMockForAbstractClass(AbstractTransport::class);
```